pytorch_lightning/plugins/training_type/parallel.py

```diff
@@ -40,13 +40,17 @@ def __init__(
 
     @property
     @abstractmethod
-    def root_device(self):
+    def root_device(self) -> torch.device:
         raise NotImplementedError
 
     @property
-    def on_gpu(self):
+    def on_gpu(self) -> bool:
         return self.root_device.type == "cuda" and torch.cuda.is_available()
 
+    @property
+    def on_tpu(self) -> bool:
+        return False
+
```

Comment:
@awaelchli why do we use `DeviceType` elsewhere in the accelerator connector? is that because that's the "intended" device type while this plugin represents what the currently set device is? should we try to unify this later, or are they serving different purposes? 


https://github.com/PyTorchLightning/pytorch-lightning/blob/20f63377f81f4771d3f128f979b3a0f9b8d219a7/pytorch_lightning/trainer/connectors/accelerator_connector.py#L539-L541

in the meantime, @@kaushikb11 can we derive `on_tpu` based off `root_device` ?

Reply:
good point, modified `on_tpu` to check based on `root_device` and whether `_XLA_AVAILABLE`

Reply:
> @awaelchli why do we use DeviceType elsewhere in the accelerator connector? is that because that's the "intended" device type while this plugin represents what the currently set device is?

I don't know exactly, I didn't add it but something like that. For internal use, to determine which accelerator to use.